src/epics/timeline.js

```diff
@@ -0,0 +1,66 @@
+import { Observable } from 'rxjs';
+import { Map, OrderedMap } from 'immutable';
+import * as TimelineActions from 'app/actions/timeline';
+import RevisionEntity from 'app/entities/revision';
+
+const getWikiDomain = ( wiki, wikis ) => {
+	let wikiRecord = wikis.find( ( obj ) => {
+		return obj.get( 'id' ) === wiki;
+	} );
+
+	return wikiRecord.domain;
+};
+
+const buildUrl = ( store ) => {
+	const domain = getWikiDomain( store.getState().query.wiki, store.getState().wikis );
+	const users = store.getState().query.user.join( '|' );
+	const ucstart = store.getState().query.startDate;
+
+	let url = 'https://' + domain + '/w/api.php?action=query&list=usercontribs&ucuser=' + users + '&ucstart=' + ucstart + '&uclimit=500&ucdir=older&format=json&origin=*';
+
+	return url;
+};
+
+const fetchTimeline = ( action$, store ) => (
+	action$.filter( ( action ) => {
+		return [ 'QUERY_UPDATE', 'QUERY_SET_VALUE', 'WIKIS_SET' ].includes( action.type );
+	} )
+		.filter( () => {
+			return !!store.getState().wikis.size;
+		} )
+		.filter( () => {
+			return !( store.getState().query.user.size < 2 || !store.getState().query.startDate );
```

Comment:
This isn't what is required... technically... the only thing that is required is wiki (as in `query.wiki`) and 1 user. The start date and end date are technically always optional.